pkg/build/admission/jenkinsbootstrapper/admission.go

```diff
@@ -0,0 +1,159 @@
+package jenkinsbootstrapper
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	coreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	authenticationclient "github.com/openshift/origin/pkg/auth/client"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	jenkinscontroller "github.com/openshift/origin/pkg/build/controller/jenkins"
+	"github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/config/cmd"
+)
+
+func init() {
+	admission.RegisterPlugin("JenkinsBootstrapper", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewJenkingsBootstrapper(c.Core()), nil
+	})
+}
+
+type jenkingsBootstrapper struct {
+	*admission.Handler
+
+	privilegedRESTClientConfig restclient.Config
+	serviceClient              coreclient.ServicesGetter
+	openshiftClient            client.Interface
+
+	jenkinsConfig configapi.JenkinsPipelineConfig
+}
+
+// NewJenkingsBootstrapper returns an admission control for builds that overrides
+// settings on builds
+func NewJenkingsBootstrapper(serviceClient coreclient.ServicesGetter) admission.Interface {
+	return &jenkingsBootstrapper{
+		Handler:       admission.NewHandler(admission.Create),
+		serviceClient: serviceClient,
+	}
+}
+
+// Admit appplies configured overrides to a build in a build pod
+func (a *jenkingsBootstrapper) Admit(attributes admission.Attributes) error {
+	if a.jenkinsConfig.Enabled != nil && !*a.jenkinsConfig.Enabled {
```

Comment:
> and should we also check if the jenkinsConfig is != nil ?

`jenkinsConfig` isn't a pointer because it isn't a pointer in the master-config.